aten/src/ATen/native/cuda/BatchLinearAlgebraLib.cu

```diff
@@ -161,6 +150,157 @@ Tensor _inverse_helper_cuda_lib(const Tensor& self) {
+  auto VT_data = VT.data_ptr<scalar_t>();
+  auto self_stride = matrixStride(self);
+  auto U_stride = matrixStride(U);
+  auto S_stride = S.size(-1);
+  auto VT_stride = matrixStride(VT);
+
+  int batchsize = cuda_int_cast(batchCount(self), "batch size");
+  int m = cuda_int_cast(self.size(-2), "m");
+  int n = cuda_int_cast(self.size(-1), "n");   
+
+  TORCH_INTERNAL_ASSERT(m <= 32 && n <= 32, "gesvdjBatched requires both matrix dimensions not greater than 32, but got "
+                        "m = ", m, " n = ", n);
+
+  // gesvdj_params controls the numerical accuracy of cusolver gesvdj iterations on GPU
+  gesvdjInfo_t gesvdj_params;
+  TORCH_CUSOLVER_CHECK(cusolverDnCreateGesvdjInfo(&gesvdj_params));
+  // TORCH_CUSOLVER_CHECK(cusolverDnXgesvdjSetTolerance(gesvdj_params, 1.0e-7));
+  // TORCH_CUSOLVER_CHECK(cusolverDnXgesvdjSetMaxSweeps(gesvdj_params, 15));
+  TORCH_CUSOLVER_CHECK(cusolverDnXgesvdjSetSortEig(gesvdj_params, 1));
+
+  auto handle = at::cuda::getCurrentCUDASolverDnHandle();
+  auto jobz = compute_uv ? CUSOLVER_EIG_MODE_VECTOR : CUSOLVER_EIG_MODE_NOVECTOR;
+  at::cuda::solver::gesvdjBatched<scalar_t>(
+    handle, jobz, m, n, self_data, m, S_data, U_data, m, VT_data, n,
+    infos.data_ptr<int>(), gesvdj_params, batchsize
+  );
+
+  TORCH_CUSOLVER_CHECK(cusolverDnDestroyGesvdjInfo(gesvdj_params));
+}
+
+// wrapper around _apply_svd_lib_gesvdjBatched that handles dtype dispatch,
+// creates a working copy of the input, and creates V^H from the V returned by gesvdj
+inline static void apply_svd_lib_gesvdjBatched(const Tensor& self, Tensor& U, Tensor& S, Tensor& VT, Tensor& infos, bool compute_uv) {
+  const int64_t m = self.size(-2);
+  const int64_t n = self.size(-1);
+  Tensor self_working_copy = cloneBatchedColumnMajor(self);
+  VT = VT.transpose(-2, -1);  // gesvdj returns V instead of V^H
+
+  AT_DISPATCH_FLOATING_AND_COMPLEX_TYPES(self.scalar_type(), "svd_cuda_gesvdjBatched", [&] {
+    _apply_svd_lib_gesvdjBatched<scalar_t>(self_working_copy, U, S, VT, infos, compute_uv);
+  });
+
+  VT = VT.conj();
+}
+
+// entrance of calculations of `svd` using cusolver gesvdj and gesvdjBatched
+std::tuple<Tensor, Tensor, Tensor> _svd_helper_cuda_lib(const Tensor& self, bool some, bool compute_uv) {
+  const int64_t batch_size = batchCount(self);
+  at::Tensor infos = at::zeros({batch_size}, self.options().dtype(at::kInt));
+  const int64_t m = self.size(-2);
+  const int64_t n = self.size(-1);
+  const int64_t k = std::min(m, n);
+
+  char jobchar = compute_uv ? (some ? 'S' : 'A') : 'N';
+
+  Tensor U_working_copy, S_working_copy, VT_working_copy;
+  std::tie(U_working_copy, S_working_copy, VT_working_copy) = \
+    _create_U_S_VT(self, some, compute_uv, /* svd_use_cusolver = */ true);
+  // U, S, V working copies are already column majored now
+
+  if (self.numel() > 0) {
+    // heuristic for using `gesvdjBatched` over `gesvdj`
+    if (m <= 32 && n <= 32 && batch_size > 1 && (!some || m == n)) {
```

Comment:
@ngimel asks: should this be `some` instead of `!some`? A comment elaborating on this check would help.

Reply:
It follows the same heuristic here in tensorflow

https://github.com/tensorflow/tensorflow/blob/09d8e7548c1b56fa2b7d3399cec51cbd9543f533/tensorflow/core/kernels/linalg/svd_op_gpu.cu.cc#L118-L119

Reply:
From the documentation, it seems that `gesvdjBatched` is designed to work only for `some=False` or `full_matrices=True`. I think it's worth adding a comment in the code about that.